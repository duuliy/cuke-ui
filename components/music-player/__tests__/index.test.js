import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MusicPlayer from "../index";

const audioLists = [
  {
    name: "星球坠落",
    singer: "中国新说唱",
    musicSrc: "http://mp3.flash127.com/public/t/id/45169.html",
    cover:
      "http://p1.music.126.net/4k-pMEO-en8IE6PdJoAYfg==/109951163429466895.jpg?param=130y130"
  }
];

describe("<MusicPlayer/>", () => {
  it("should render a <MusicPlayer/> components", () => {
    const wrapper = render(<MusicPlayer audioLists={audioLists} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-music-player classnames", () => {
    const wrapper = shallow(<MusicPlayer audioLists={audioLists} />);
    assert(wrapper.find(".cuke-music-player").length === 1);
  });
});
