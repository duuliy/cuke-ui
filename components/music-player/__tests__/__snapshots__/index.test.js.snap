// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MusicPlayer/> should render a <MusicPlayer/> components 1`] = `
<div
  class="react-jinke-music-player-main cuke-music-player"
>
  <div
    class="react-jinke-music-player react-draggable"
    style="touch-action:none;left:0;top:0;transform:translate(0px,0px)"
  >
    <div
      class="music-player"
    >
      <div
        class="scale music-player-controller"
        id="music-player-controller"
        style="background-image:url(http://p1.music.126.net/4k-pMEO-en8IE6PdJoAYfg==/109951163429466895.jpg?param=130y130)"
      >
        <span
          class="controller-title"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align:middle"
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m38.6 19.8q0 3.7-1.3 7l-0.4 1.1-4.2 0.7q-0.5 1.9-2 3.1t-3.5 1.2v0.7q0 0.3-0.2 0.5t-0.5 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5v0.7q1.6 0 2.9 0.8t2.1 2.2l1.5-0.3q0.7-2.1 0.7-4.3 0-3.3-2-6.3t-5.3-4.6-7-1.8-7.1 1.8-5.2 4.6-2 6.3q0 2.2 0.6 4.3l1.6 0.2q0.7-1.3 2-2.1t2.9-0.8v-0.7q0-0.3 0.2-0.5t0.5-0.2h1.5q0.3 0 0.5 0.2t0.2 0.5v12.9q0 0.3-0.2 0.5t-0.5 0.2h-1.5q-0.3 0-0.5-0.2t-0.2-0.5v-0.7q-1.9 0-3.5-1.2t-2-3.1l-4.1-0.7-0.5-1.1q-1.3-3.3-1.3-7 0-3.4 1.5-6.5t4-5.4 5.9-3.7 7.2-1.3 7.1 1.3 6 3.7 4 5.4 1.5 6.5z"
              />
            </g>
          </svg>
        </span>
        <div
          class="music-player-controller-setting"
        >
          打开
        </div>
      </div>
    </div>
  </div>
  <div
    class="audio-lists-panel"
  >
    <div
      class="audio-lists-panel-header"
    >
      <h2
        class="title"
      >
        <span>
          播放列表 / 
        </span>
        <span
          class="num"
        >
          1
        </span>
        <span
          class="close-btn"
          title="Close"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align:middle"
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
              />
            </g>
          </svg>
        </span>
        <span
          class="line"
        />
        <span
          class="delete-btn"
          title="Delete audio lists"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style="vertical-align:middle"
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m15.9 16.4v12.9q0 0.3-0.2 0.5t-0.5 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m5.7 0v12.9q0 0.3-0.2 0.5t-0.5 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m5.8 0v12.9q0 0.3-0.2 0.5t-0.6 0.2h-1.4q-0.3 0-0.5-0.2t-0.2-0.5v-12.9q0-0.3 0.2-0.5t0.5-0.2h1.4q0.4 0 0.6 0.2t0.2 0.5z m2.8 16.2v-21.2h-20v21.2q0 0.5 0.2 0.9t0.3 0.6 0.2 0.2h18.6q0.1 0 0.2-0.2t0.4-0.6 0.1-0.9z m-15-24h10l-1.1-2.6q-0.1-0.2-0.3-0.3h-7.1q-0.2 0.1-0.4 0.3z m20.7 0.7v1.4q0 0.3-0.2 0.5t-0.5 0.2h-2.1v21.2q0 1.8-1.1 3.2t-2.5 1.3h-18.6q-1.4 0-2.5-1.3t-1-3.1v-21.3h-2.2q-0.3 0-0.5-0.2t-0.2-0.5v-1.4q0-0.3 0.2-0.5t0.5-0.2h6.9l1.6-3.8q0.3-0.8 1.2-1.4t1.7-0.5h7.2q0.9 0 1.8 0.5t1.2 1.4l1.5 3.8h6.9q0.3 0 0.5 0.2t0.2 0.5z"
              />
            </g>
          </svg>
        </span>
      </h2>
    </div>
    <div
      class="audio-lists-panel-content"
    >
      <div
        role="presentation"
      >
        <ul>
          <li
            class="audio-item playing pause"
            title="Click to play"
          >
            <span
              class="group player-status"
            >
              <span
                class="player-icons"
              >
                <svg
                  class="react-jinke-music-player-play-icon"
                  fill="currentColor"
                  height="1em"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align:middle"
                  viewBox="0 0 40 40"
                  width="1em"
                >
                  <g>
                    <path
                      d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6-2.3 8.6-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3z m8.6 18.3q0.7-0.4 0.7-1.2t-0.7-1.2l-12.1-7.2q-0.7-0.4-1.5 0-0.7 0.4-0.7 1.3v14.2q0 0.9 0.7 1.3 0.4 0.2 0.8 0.2 0.3 0 0.7-0.2z"
                    />
                  </g>
                </svg>
              </span>
            </span>
            <span
              class="group player-name"
            >
              星球坠落
            </span>
            <span
              class="group player-time"
            >
              中国新说唱
            </span>
            <span
              class="group player-delete"
              title="Click to delete 星球坠落"
            >
              <svg
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align:middle"
                viewBox="0 0 40 40"
                width="1em"
              >
                <g>
                  <path
                    d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
                  />
                </g>
              </svg>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <audio
    class="music-player-audio"
    src="http://mp3.flash127.com/public/t/id/45169.html"
  />
</div>
`;
